rfd/0067-database-access-aws-redis

```diff
@@ -0,0 +1,276 @@
+---
+authors: STeve Huang (xin.huang@goteleport.com)
+state: draft
+---
+
+# RFD 67 - Redis database access in AWS ElastiCache and MemoryDB
+
+## Required Approvers
+* Engineering: @r0mant && @smallinsky
+* Product: (@klizhentas || @xinding33)
+* Security: @reedloden
+
+## What
+
+Support Redis database access in AWS
+[ElastiCache](https://aws.amazon.com/elasticache/redis/) and
+[MemoryDB](https://aws.amazon.com/memorydb/).
+
+## Why
+
+Redis is one of the most popular in-memory databases, and Amazon offers AWS
+managed solutions like ElastiCache and MemoryDB.
+
+As described in [Database Access
+RFD](https://github.com/gravitational/teleport/blob/master/rfd/0011-database-access.md),
+mutual TLS is used between database clients and proxy, and between proxy and
+database service. The same applies to Redis access.
+
+However, neither mutual TLS nor IAM authentication is supported by Redis
+managed by AWS. This document mainly focuses on access from database service to
+Redis managed by AWS.
+
+## Details
+
+### In-transit encryption (TLS)
+
+For security purposes, Teleport should ONLY
+support databases with in-transit encryption (TLS) enabled.
+
+[Enabling TLS on ElastiCache
+Redis](https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/in-transit-encryption.html#in-transit-encryption-enable)
+can be a complicated process, and all client applications must be updated at
+the same time. Therefore, Teleport should NOT automatically enable TLS for
+existing clusters. Redis servers without TLS are NOT supported, and users are
+recommended to enable TLS separately to enable Teleport database access.
+
+### Auth
+
+Redis can be configured with one of the following authentication methods, and
+only one at a time.
+
+#### Auth - Redis with no auth
+
+Redis can be configured without any auth method.
+
+"default" user is always used in this configuration, and no `auth` command is
+required upon a successful connection.
+
+#### Auth - Redis with Redis Auth
+
+[Redis Auth](https://redis.io/commands/auth/)
+uses a single token/password that must be shared for all client applications.
+
+"default" user is always used in this configuration, and users have to manually
+input `auth` command upon a successful connection. <sup>1<sup>
+
+#### Auth - Redis with ACL
+
+[Redis ACL](https://redis.io/docs/manual/security/acl/) (also known as RBAC for
+ElastiCache Redis) is introduced in Redis 6. Both ElastiCache and MemoryDB
+provide APIs to manage database users and user groups for ACL.
+
+As with all other databases, the database users should be created by "database
+admins" (by users, not Teleport) to assign desired permissions. 
+
+To provide better security than using static passwords, users can tag
+ElastiCache and MemoryDB users with label **`teleport.dev/managed: true`** for
+Teleport to manage the passwords for those database users.
```

Comment:
It would be cool to update our configurator to do this. It could fetch the list of users and ask the admin which Redis users they would like to enroll into this rotation, or provision new users from scratch. Not in the initial implementation scope though.